hydromt/data_adapter/dataset.py

```diff
@@ -0,0 +1,431 @@
+"""Implementation for the dataset DataAdapter."""
+import logging
+from datetime import datetime
+from os.path import basename, splitext
+from pathlib import Path
+from typing import List, NewType, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from pystac import Asset as StacAsset
+from pystac import Catalog as StacCatalog
+from pystac import Item as StacItem
+from pystac import MediaType
+
+from hydromt.typing import ErrorHandleMethod, TimeRange
+
+from .data_adapter import DataAdapter
+from .utils import netcdf_writer, shift_dataset_time, zarr_writer
+
+logger = logging.getLogger(__name__)
+
+DatasetSource = NewType("DatasetSource", str | Path)
+
+
+class DatasetAdapter(DataAdapter):
+    """DatasetAdapter for non-spatial n-dimensional data."""
+
+    _DEFAULT_DRIVER = ""
+    _DRIVERS = {
+        "nc": "netcdf",
+    }
+
+    def __init__(
+        self,
+        path: str | Path,
+        driver: Optional[str] = None,
+        filesystem: Optional[str] = None,
+        nodata: Optional[dict | float | int] = None,
+        rename: Optional[dict] = None,
+        unit_mult: Optional[dict] = None,
+        unit_add: Optional[dict] = None,
+        meta: Optional[dict] = None,
+        attrs: Optional[dict] = None,
+        driver_kwargs: Optional[dict] = None,
+        storage_options: Optional[dict] = None,
+        name: Optional[str] = "",
+        catalog_name: Optional[str] = "",
+        provider: Optional[str] = None,
+        version: Optional[str] = None,
+    ):
+        """Initiate data adapter for n-dimensional timeseries data.
+
+        This object contains all properties required to read supported files into
```

Comment:
do we know which file types are supported? can we link to it somewhere? if not I think we should make a ticket for this. It would be good to add to the documentation.

Reply:
for now netcdf and zarr are supported. We still have to work on supporting multiple csv files.